feat: show person details

wroblewski82 opened the pull request:
I have implemented a dynamic Actor Details Page that updates in real-time when a user clicks on an actor's tile. Upon clicking, the page retrieves detailed information about the actor from the API using their ID. This includes information such as the actor's date and place of birth, as well as their biography. The page also displays a list of movies in which the actor has participated.

sofyahreksoftware commented on src/common/ImagePlaceholder/styled.js:
```diff
@@ -26,4 +26,15 @@ export const ImagePlaceholder = styled.div`
+    css`
+      width: 399px;
+      height: 564px;
+      @media (max-width: ${({ theme }) => theme.media.mobile}px) {
```
It's better to add a blank line here for better readability.

sofyahreksoftware commented on src/features/PeopleDetails/ActorProfile/styled.js:
```diff
@@ -0,0 +1,97 @@
+  }
+`;
+
+export const StyledDiv = styled.div`
```
I think StyledDiv isn't a good name:) What do you think about naming it as something like`**DataContainer**`?🤔

sofyahreksoftware commented on src/features/PeopleDetails/MoviesList/index.js:
```diff
@@ -0,0 +1,31 @@
+      <StyledList>
+        {moviesList.map((movie) => (
+          <StyledItem key={movie.id}>
+            <MovieTile
+              poster={`https://image.tmdb.org/t/p/w342/${movie.poster_path}`}
+              title={movie.title}
+              year={movie.release_date.split("-")[0]}
+              genres={movie.genre_ids.map((id) =>
+                genreList.find((genre) => genre.id === id)
+              )}
+              mark={movie.vote_average.toFixed(1)}
+              votes={movie.vote_count}
+            />
+          </StyledItem>
```
What do you think about adding `*?*` operator before methods as _toFixed/map_ etc. This will avoid potential errors:)

wroblewskisoftware replied:
I think it's a good idea.

sofyahreksoftware reviewed (changes requested):
Good job! As always, I have a few tips:)

- I think there is no point in adding a MovieList component with styles in the PeopleDetails folder. We already have a MovieList component with its styles and HTML skeleton as a separate feature, and we also have a styled.js file in features with styled lists. You can use styles imported directly from there.

-Additionally, it would be better if you rename `**PeopleDetailsSaga**` and `**PeopleDetailsSlice**` to `**PersonDetailsSaga**` and `**PersonDetailsSlice**` because in this situation we manage information about one person. Furthermore, we should decide on one name: Actor or Person. I think `*Person*` would be better because we represent information not only about actors but also directors, cast, and crew, etc.

- It's better to rename ActorProfile component to **PersonPage** for consistency with the remaining features' names of the App. And also change names of related names (actor to person):

I have also left some notes in comments directly in code, so bring attention to them:)

Congratulation! It was a good job and and good contribution to the development of the application🤪😍

wroblewski82 commented:
Hi Sofia!
Thanks for your feedback. Here are the changes I made:

1. Styling the MoviesList component:
I moved the StyledHeader to the general styles, using props and changing the tag. This should improve style consistency.

2. Names PeopleDetailsSaga, PeopleDetailsSlice, and PeopleDetails:
I kept these names because they refer to multiple people, not single cases. This makes the code more intuitive.

3. Distinction between people and person:
In some places, I used "people" and in others "person" to better differentiate the context of these names.

4. Changing ActorProfile to PersonPage:
I changed the name ActorProfile to PersonPage, which better reflects the content of the page.

5. Minor adjustments:
I made a few minor adjustments to improve the functionality and readability of the code.

I believe these changes are reasonable, and the code can stay in this form.
Thanks for your work and suggestions!

Best regards,
Radek

sofyahreksoftware reviewed (approved):
Now it looks better.
There is also a small change you can make.
Before all the attributes in the MovieList and PersonPage files, you can add a dollar sign ($) so that the code is consistent with the latest practices. You can read more about it in the React documentation here.
https://styled-components.com/docs/api#transient-props
